packages/story-editor/src/app/taxonomy/taxonomyProvider.tsx

```diff
@@ -56,6 +62,7 @@ function TaxonomyProvider(props) {
     })
   );
 
+  // @ts-ignore Reason: update this after useAPI is updated ?
```

Comment:
This should now be unblocked, #12673 was merged

Reply:
Doesn't seem to like the overloaded function
```
function useAPI(): APIState;
function useAPI<T>(selector: (state: APIState) => T | APIState = identity) {
  return useContextSelector(Context, selector ?? identity);
}
```

<img width="600" alt="Screen Shot 2022-11-21 at 7 32 13 AM" src="https://user-images.githubusercontent.com/62242/203056116-03340ddf-d40d-473e-bfd1-269239b950f4.png">

Testing without the overload 

```
function useAPI<T>(selector: (state: APIState) => T | APIState = identity) {
  return useContextSelector(Context, selector ?? identity);
}
```


<img width="1231" alt="Screen Shot 2022-11-21 at 7 31 47 AM" src="https://user-images.githubusercontent.com/62242/203056175-1f52f918-4b61-4881-a5d4-708e63b3fdfa.png">

Reply:
Looks like we might need to add this overload as well:
`function useAPI<T>(selector: (state: APIState) => T): T;`

Does it work when adding that?